src/Build/Evaluation/Evaluator.cs

```diff
@@ -1790,8 +1791,33 @@ private List<ProjectRootElement> ExpandAndLoadImports(string directoryOfImportin
                 if (solutionPath == "*Undefined*") solutionPath = null;
                 var projectPath = _data.GetProperty(ReservedPropertyNames.projectFullPath)?.EvaluatedValue;
 
+                var compareInfo = CultureInfo.InvariantCulture.CompareInfo;
+
+                static bool HasProperty(string value, CompareInfo compareInfo) =>
+                    value != null && compareInfo.IndexOf(value, "$(") != -1;
+
+                if (HasProperty(sdkReference.Name, compareInfo) ||
+                    HasProperty(sdkReference.Version, compareInfo) ||
+                    HasProperty(sdkReference.MinimumVersion, compareInfo))
+                {
+                    static string EvaluateProperty(string value, IElementLocation location, Expander<P, I> expander) =>
+                        value != null
+                            ? expander.ExpandIntoStringAndUnescape(value,
+                                                                   ExpanderOptions.ExpandProperties,
+                                                                   location)
+                            : null;
+
+                    var sdkReferenceOrigin = importElement.SdkLocation;
+
+                    sdkReference = new SdkReference(
```

Comment:
Before you had a check to only create a `new SdkReference` if expansion changed the contents of the original sdk reference string. I think that's a sensible check to keep.

Reply:
The check below that I've removed only influenced logging the event. It didn't influence overriding `sdkReference` variable. When `sdkReference` contains something that can theoretically be expanded (`$(` heuristic above), it is always created anew from the result of the expansion.

Reply:
Ah, yes, good point.